### PR TITLE
[FEATURE] Server: Add handling for custom middleware

### PR DESCRIPTION
**Pull request by RandomByte**
As per RFC 0005: https://github.com/SAP/ui5-tooling/pull/151

### CODE REVIEW

File: lib/middleware/discovery.js [hunk 1]
```diff
@@ -13,12 +13,14 @@ const urlPattern = /\/(app_pages|all_libs|all_tests)(?:[?#].*)?$/;
...
  * @module @ui5/server/middleware/discovery
- * @param {Object} resourceCollections Contains the resource reader or collection to access project related files
- * @param {module:@ui5/fs.AbstractReader} resourceCollections.source Resource reader or collection for the source project
- * @param {module:@ui5/fs.AbstractReader} resourceCollections.combo Resource collection which contains the workspace and the project dependencies
+ * @param {Object} parameters Parameters
+ * @param {module:@ui5/fs.AbstractReader} parameters.resources.all Reader or Collection to read resources of the
+ *                                        root project and its dependencies
+ * @param {module:@ui5/fs.AbstractReader} parameters.resources.rootProject Reader or Collection to read resources of
+ *                                        the project the server is started in
  * @returns {Function} Returns a server middleware closure.
```
Comment (matz3) [inline lib/middleware/discovery.js]: Signature changed to Promise<function>, right?

Comment (RandomByte) [reply lib/middleware/discovery.js]: Maybe we could make that optional? https://github.com/SAP/ui5-tooling/pull/151/commits/81d4add776b8c91e214b7d9ed0131a55bebadeff

**Comment by RandomByte** [comment]
We should consider releasing this as a 2.0 since the index.js and the signatures of the middleware modules changed significantly.

**Comment by matz3** [comment]
We could also deprecate the old middleware (exported via index.js) and map the argument names, right?
This way we wouldn't need to make a 2.0 and give consumers a chance to adopt their code in advance (although we never documented the exported middleware as public API).

**Comment by RandomByte** [comment]
> We could also deprecate the old middleware (exported via index.js) and map the argument names, right?

Done in latest change

**Pull request merged**